test/TypeCheck/Positive.hs

```diff
@@ -39,6 +39,10 @@ tests =
       "Literal String matches any type"
       "MicroJuvix"
       "LiteralString.mjuvix",
+    PosTest
+      "Box type"
```

Comment:
Is there anything special about this example I don't see? why should we have it?

Reply:
Nothing special. It is just a simple example that's easy to debug. It helped me debug some of the errors I was having when working in this PR. I thought there was no harm in adding it to the test suite.